src/classes/BDI_AdditionalObjectService.cls

```diff
@@ -279,7 +291,90 @@ public with sharing class BDI_AdditionalObjectService {
 
 
     /*******************************************************************************************************
-    * @description Determines if the predecessor to this Data Import Object Mapping exists
+    * @description For newly created parent sobjects, this method updates the appropriate child predecessor
+    * record's lookup field to the id of the newly created object.
+    * @param objWrapsForChildPredecessorUpdate list of object wraps that were successfully created
+    * and need to update their child predecessors.
+    */
+    private void updateChildPredecessorRecords(BDI_ObjectWrapper[] objWrapsForChildPredecessorUpdate){
+
+        Map<String,Map<Id,SObject>> childSObjectMapByObjectAPIName = new Map<String,Map<Id,SObject>>();
+        Map<Id,BDI_ObjectWrapper[]> successorObjWrapsByPredecessorId = new Map<Id,BDI_ObjectWrapper[]>();
+
+        for (BDI_ObjectWrapper objWrap : objWrapsForChildPredecessorUpdate) {
+
+            Id predecessorId = (Id)objWrap.dataImport.get(objWrap.predecessorObjMapping.Imported_Record_Field_Name__c);
+            SObject sObjForUpdate;
+
+            Map<Id,SObject> tempSObjMap = new Map<Id,SObject>();
```

Comment:
Even though this is internal to this loop, can we rename this "sObjectById" just for clarity of what's being stored? Or even tempSObjectById if we want to label it as a temporary collection

Reply:
updated the name.